os/include/tinyara/sched.h

```diff
@@ -866,6 +866,66 @@ void sched_foreach(sched_foreach_t handler, FAR void *arg);
  */
 FAR struct tcb_s *sched_gettcb(pid_t pid);
 
+#ifdef CONFIG_SCHED_SAVE_DEADTASK
+
+/****************************************************************************
+ * Name: sched_deadtasklistinit
+ *
+ * Description:
+ *   This function initializes dead task info list
+ *
+ * Inputs:
+ *   None
+ *
+ ****************************************************************************/
+
+void sched_deadtasklistinit();
+
+/****************************************************************************
+ * Name: sched_addDeadtask
+ *
+ * Description:
+ *   This function adds a task's pid and name info to the list.
+ *   This should be called before deleting a task.
+ *
+ * Inputs:
+ *   pid: pid of the deleting task
+ *   name: name of the task
+ *
+ ****************************************************************************/
+
+void sched_addDeadtask(int pid, char *name);
+
+/****************************************************************************
+ * Name: sched_removedeadtask
+ *
+ * Description:
+ *   This function removes a task's pid and name info from dead list
+ *   if present. This should be called after creating a task.
```

Comment:
Pls change as below. 
This api must be called at the time of creation of a new task, in order to make sure that we dont store any task name with same pid as the new task in the dead task list.